chore: updated the publishing documentation (#464)
pricelinelabs/design-system: CONTRIBUTING.md

To publish the packages to npm, you'll need to be added as an owner for the
packages you're publishing. Use the #design-system Slack channel for more
information. 

Before Publishing:

> - Know that publishing is very easy once you have access to do so **Please Be Careful** 🤗

Reviewer: A tiny bit more concise?

```suggestion
- Publishing is very easy once you have access to the NPM package **Please Be Careful** 🤗
```

> - As of v2 you must be on Node v8+

Reviewer: ```suggestion
- As of v2, Node.js v8+ is required
```

> - name sure you are logged into NPM properly, if you have multiple logins you can use something like [npmrc]

Reviewer: ```suggestion
- Ensure NPM login has been authenticated. If multiple NPM registries are used, [npmrc]() tool can be used to switch to NPM public registry.
```

> - name sure you are logged into NPM properly, if you have multiple logins you can use something like [npmrc]

Reviewer: > [npmrc]() ...

Would you like to provide the npmrc variable, and the npmrc link at the bottom? (If you prefer a follow-up PR, that's cool)

These Lerna commands can be helpful when publishing:

- Run `npm run changed` to see which packages have changed since the last
  release.
- Run `npm run diff` to see a diff of all packages since the last release.

> - Run `npm run publish` to publish the updated packages with Lerna.

Reviewer: Would you mind removing the periods of these bullets, before the last bullet? (think it'd be consistent with the format of other bullet list.)

> - Do not run `npm publish` on it's own.

Reviewer: ```suggestion
-- Do not run `npm publish`, since `npm run publish` script will use `lerna publish` to manage publish multiple packages.
```

> To set the packages to the same SemVer/custom tag run `npm run publish` and follow the prompts from the lerna-cli.  

Reviewer: ```suggestion
To set the packages to the same SemVer/custom tag, run `npm run publish` and follow the prompts from the lerna-cli.  
```

> To set the packages to the same SemVer/custom tag run `npm run publish` and follow the prompts from the lerna-cli.  

Reviewer: Tiny nit: if you like to break line when > 80 characters, think you can start with a space in 2nd line in markdown, and the formatted page would still display the line without breaking.

> To set the SemVer for each package individually run `npm run publish -- --independent` and the lerna-cli will prompt and ask the SemVer/custom tag for each package individually.

Reviewer: ```suggestion
To set the SemVer for each package individually, run `npm run publish -- --independent. lerna-cli will prompt, and ask the SemVer/custom tag for each package individually.
```
